lighthouse-core/audits/legacy-javascript.js

```diff
@@ -0,0 +1,344 @@
+      // e(e.S,"Object",{values
+      // minified pattern found in babel-polyfill
+      // see https://cdnjs.cloudflare.com/ajax/libs/babel-polyfill/7.2.5/polyfill.min.js
+      expression += `|;e\\([^,]+,${qt(objectWithoutPrototype)},{${property}`;
+    }
+
+    return expression;
+  }
+
+  /**
+   * @return {Pattern[]}
+   */
+  static getPolyfillPatterns() {
+    return [
+      'Array.prototype.filter',
+      'Array.prototype.find',
+      'Array.prototype.findIndex',
+      'Array.prototype.forEach',
+      'Array.from',
+      'Array.prototype.indexOf',
+      'Array.isArray',
+      'Array.prototype.lastIndexOf',
+      'Array.prototype.map',
+      'Array.of',
+      'Array.prototype.reduce',
+      'Array.prototype.reduceRight',
+      'Array.prototype.some',
+      'Date.now',
+      'Date.prototype.toISOString',
+      'Date.prototype.toJSON',
+      'Date.prototype.toString',
+      'Function.prototype.bind',
+      'Function.prototype.name',
+      'Map',
+      'Number.isInteger',
+      'Number.isSafeInteger',
+      'Number.parseFloat',
+      'Number.parseInt',
+      'Object.assign',
+      'Object.create',
+      'Object.defineProperties',
+      'Object.defineProperty',
+      'Object.freeze',
+      'Object.getOwnPropertyDescriptor',
+      'Object.getOwnPropertyNames',
+      'Object.getPrototypeOf',
+      'Object.isExtensible',
+      'Object.isFrozen',
+      'Object.isSealed',
+      'Object.keys',
+      'Object.preventExtensions',
+      'Object.seal',
+      'Object.setPrototypeOf',
+      'Promise',
+      'Reflect.apply',
+      'Reflect.construct',
+      'Reflect.defineProperty',
+      'Reflect.deleteProperty',
+      'Reflect.get',
+      'Reflect.getOwnPropertyDescriptor',
+      'Reflect.getPrototypeOf',
+      'Reflect.has',
+      'Reflect.isExtensible',
+      'Reflect.ownKeys',
+      'Reflect.preventExtensions',
+      'Reflect.set',
+      'Reflect.setPrototypeOf',
+      'Set',
+      'String.prototype.codePointAt',
+      'String.prototype.endsWith',
+      'String.fromCodePoint',
+      'String.prototype.includes',
+      'String.raw',
+      'String.prototype.repeat',
+      'String.prototype.startsWith',
+      'String.prototype.trim',
+      'ArrayBuffer',
+      'DataView',
+      'Float32Array',
+      'Float64Array',
+      'Int16Array',
+      'Int32Array',
+      'Int8Array',
+      'Uint16Array',
+      'Uint32Array',
+      'Uint8Array',
+      'Uint8ClampedArray',
+      'WeakMap',
+      'WeakSet',
+      'Array.prototype.includes',
+      'Object.entries',
+      'Object.getOwnPropertyDescriptors',
+      'Object.values',
+      'String.prototype.padEnd',
+      'String.prototype.padStart',
+    ].map(str => {
+      const parts = str.split('.');
+      const object = parts.length > 1 ? parts.slice(0, parts.length - 1).join('.') : null;
+      const property = parts[parts.length - 1];
+      return {
+        name: `${object ? object + '.' : ''}${property}`,
```

Comment:
isn't this just piecing `str` back together again or am I missing something?

Reply:
yeah.... um..... lol